dapr/clients/grpc/client.py

```diff
@@ -461,6 +462,73 @@ def get_bulk_state(
             items=items,
             headers=call.initial_metadata())
 
+    def query_state_alpha1(
+            self,
+            store_name: str,
+            query: str,
+            states_metadata: Optional[Dict[str, str]] = dict()) -> QueryResponse:
+        """Queries a statestore with a query
+
+        For details on supported queries see https://docs.dapr.io/
+
+        This example queries a statestore:
+            from dapr import DaprClient
+
+            query = '''
+            {
+                "filter": {
+                    "EQ": { "value.state": "CA" }
+                },
+                "sort": [
+                    {
+                        "key": "value.person.id",
+                        "order": "DESC"
+                    }
+                ]
+            }
+            '''
+
+            with DaprClient() as d:
+                resp = d.query_state_alpha1(
+                    store_name='state_store',
+                    query=query,
+                    states_metadata={"metakey": "metavalue"},
+                )
+
+        Args:
+            store_name (str): the state store name to query
+            query (str): the query to be executed
+            states_metadata (Dict[str, str], optional): custom metadata for state request
+
+        Returns:
+            :class:`QueryStateResponse` gRPC metadata returned from callee,
+                pagination token and results of the query
+        """
+
+        if not store_name or len(store_name) == 0 or len(store_name.strip()) == 0:
+            raise ValueError("State store name cannot be empty")
+        req = api_v1.QueryStateRequest(
+            store_name=store_name,
+            query=query,
+            metadata=states_metadata)
+        response, call = self._stub.QueryStateAlpha1.with_call(req)
```

Comment:
Q do we ever do async await patterns for SDK calls that could take a while?

Reply:
Actually that seems out of scope but requested here: https://github.com/dapr/python-sdk/issues/90

Reply:
I'm keeping this consistent for now. Making everything async/await will be a different effort. It also makes testing a little harder :)